cmake/SetupCompilerOptions.cmake

```diff
@@ -268,7 +268,7 @@ if (BLT_CXX_STD)
         blt_append_custom_compiler_flag(
             FLAGS_VAR CMAKE_CXX_FLAGS
             DEFAULT " "
-            XL "-std=c++1y"
+            XL "-std=c++14"
```

Comment:
@davidbeckingsale -- presumably this is here b/c xl did not previously support the `c++14` flag. 
Do you know which version of XL added support for this?

Bonus points: Do we know if there is a version of CMake that generates the right flag for XL?

Reply:
Looks like (~2 years ago) CMake 3.10 and XL 16 was still using `c++1y`
https://gitlab.kitware.com/cmake/cmake/-/issues/18071

Reply:
If I type `man xlC` on our latest version of `xl`, I also still seeing `c++1y` but not `c++14`.

Here's the relevant section:
```
> ml XL/2021.05.19   (latest)
>man xlC
 ...
-std[=<suboption>]
-qlanglvl[=<suboption>]
	   Determines whether source code and compiler options should be checked for conformance to a specific language standard, or subset or superset of a standard.
	   ...

	   (C++) For C++ language programs, the suboptions of -std include:
	   c++98 | c++03
	   gnu++98 | gnu++03
	   c++11 | c++0x
	   gnu++11 | gnu++0x
	   c++1y
```

Reply:
I think their documentation might be slightly out-of-date. Although the page below is from September 2020, so the `-std=c++14` flag must be a fairly recent change.

https://www.ibm.com/support/pages/c14-language-standard-fully-supported-xl-cc-linux-1611-fix-pack-8